### PR TITLE
[sweet API] Throw missing exceptions

### PR DESCRIPTION
**Pull request by lukmccall**
# Why

I forgot to throw exceptions in two corner cases:
- when js type is not convertible to a Kotlin object - like Symbol
- when the js value has an unknown type

# How

- Exported `UnknownException` to the C++.
- Add missing exceptions.

# Test Plan

- unit tests that will be added in a separate PR

### CODE REVIEW

File: packages/expo-modules-core/android/src/main/cpp/MethodMetadata.cpp [hunk 1]
```diff
@@ -150,8 +150,11 @@ std::vector<jvalue> MethodMetadata::convertJSIArgsToJNI(
           react::ReadableNativeMap::createWithContents(std::move(dynamic)).release());
       }
     } else {
-      // TODO(@lukmccall): throw an exception
-      jarg->l = nullptr;
+      auto stringRepresentation = arg->toString(rt).utf8(rt);
+      jni::throwNewJavaException(
+        UnexpectedException::create(
+          "Cannot convert " + stringRepresentation + " to a Kotlin type.").get()
```
Comment (tsapeta) [inline packages/expo-modules-core/android/src/main/cpp/MethodMetadata.cpp:156]: Let's add single quotes when we refer to strings or in general dynamic/interpolated values
```suggestion
          "Cannot convert '" + stringRepresentation + "' to a Kotlin type.").get()
```

File: packages/expo-modules-core/android/src/main/java/expo/modules/kotlin/exception/CodedException.kt [hunk 2]
```diff
@@ -159,7 +165,7 @@ class JavaScriptEvaluateException(
   val jsStack: String
 ) : CodedException(
   message = """
-  Cannot evaluate JavaScript code: $message.
+  Cannot evaluate JavaScript code: $message${if (message.lastOrNull() == '.') "" else "."}
```
Comment (tsapeta) [inline packages/expo-modules-core/android/src/main/java/expo/modules/kotlin/exception/CodedException.kt:168]: I thought that we decided not to end error messages with a `.` 😅

**Review by tsapeta — APPROVED** [review]
